storage/badger/badger.go

```diff
@@ -0,0 +1,85 @@
+package badger
+
+import (
+	v3Badger "github.com/dgraph-io/badger/v3"
+	"github.com/pastelnetwork/go-commons/errors"
+	"github.com/pastelnetwork/walletnode/storage"
+
+	"github.com/pastelnetwork/go-commons/log"
+)
+
+// DB structure represents wrapper over the badger.DB
+type DB struct {
+	db     *v3Badger.DB
+	logger *Logger
+}
+
+// NewDB creates an instance of BadgerDB and starts fetching data
+func NewDB(cfg *Config) storage.KeyValue {
+	ch := &DB{
+		db:     &v3Badger.DB{},
+		logger: NewLogger(),
+	}
+	if err := ch.start(cfg); err != nil {
+		ch.logger.Errorf("error caused when trying to start badger db from %v", cfg.Dir)
+		return nil
+	}
+	return ch
+}
+
+func (db *DB) Init() error {
+	return nil
+}
+
+// Init method opens the v3Badger db instance using specified configuration data and logger instance
+func (db *DB) start(cfg *Config) (err error) {
+	db.db, err = v3Badger.Open(v3Badger.DefaultOptions(cfg.Dir).WithLogger(log.DefaultLogger))
```

Comment:
still `log.DefaultLogger` ? please be more careful, this the third time I point to this place